compiler/qsc/src/error.rs

```diff
@@ -1,56 +1,40 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use miette::{Diagnostic, SourceCode};
-use qsc_frontend::compile::{Source, SourceMap};
+use miette::Diagnostic;
 use std::{
     error::Error,
     fmt::{self, Debug, Display, Formatter},
 };
+use thiserror::Error;
 
-#[derive(Clone, Debug)]
-pub(super) struct WithSource<S, E> {
-    source: Option<S>,
+#[derive(Clone, Debug, Error)]
+pub struct WithStack<E>
```

Comment:
This is just a refactoring. `WithStack` adds a `stack_trace` to an object, which is something that used to be part of the old `WithSource` struct. Only runtime errors can have stacks, so I didn't feel the need to carry around an optional stack trace field with all `WithSource` types which are usually compiler errors without traces.